ApplyToTheIndustry/Assets/Scripts/AI Grader/Grader.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Grader : MonoBehaviour
+{
+     Dictionary<string, int> skillPoints = new Dictionary<string, int>();
+     HashSet<string> countedSkills = new HashSet<string>();
+     int Totalpoints =0;
+
+    void Start()
+    {
+        
+        GetListOfSkills();
+
+    }
+
+    void Update()
+    {
+        //Checking the job type
+        if (JobContainer.Instance.currentPosting.name == "Job Posting for Programmer")
+        {
+            PointSystemForJobType1();
+
+        }
+
+        CalculateTotalPoints();
+
+        GetFeedback();
+
+    }
+
+    //Get the list of skills available
+    public void GetListOfSkills()
+    {
+        SkillGroup playerSkills = ServiceLocator.Instance.GetService<PlayerSkillsManager>().GetSkills();
+
+        foreach (Skill skill in playerSkills.ListSkills())
+        {
+            if (!skillPoints.ContainsKey(skill.Name))
+            {
+                skillPoints.Add(skill.Name, 0); // Default points are 0
+            }
+        }
+    }
+
+    //Point system for job type 1
+    public void PointSystemForJobType1()
```

Comment:
This should be fine for now, but soon we'll want to integrate this with job posting data so we dont have to hard code values.

Reply:
I agree, this seems like rather than creating this dictionary, it can be an instance of the `SkillGroup` class we've already declared.